engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -587,6 +590,7 @@ protected Answer copySnapshot(DataObject srcData, DataObject destData) {
             if (cacheData != null) {
                 cacheMgr.deleteCacheObject(cacheData);
             }
+            _snapshotService.markRemovedSnapshot(destData.getId());
```

Comment:
Can you check for OperationTimedoutException (if thrown for timeout?), set removed accordingly. Note that, the snapshot here is not in Error state as the copy callback is not yet to be called, which takes the action based on the operation failure.

Reply:
I think having the snapshot marked as removed here in the case of any exception is better than just checking for a timeout. I think if the process fails for any reason we would want to keep the db records.

Reply:
@Spaceman1984 possible to move this method call to copy snapshot callback method on result failure?

Reply:
Thanks @sureshanaparti for pointing me in that direction, I was able to get rid of that method.